### PR TITLE
Implement logic for showing gb vs mb vs kb for account resources

### LINKED ISSUE #716: Improve Resource Net and Ram Readability (display mb and gb)
## Overview
Improve readability of resources Ram and Net and any others. We should make these mb or gb... make the size human readable
## Links
telegram-cloud-photo-size-1-4940651664461900856-x

## Acceptance Criteria
1. Display mb and gb when resources reach these sizes.

### PR DESCRIPTION
**Pull request by kylanhurt**
# Fixes #716 

## Description

Max size takes precedence since it is arguably more important: used size is only relevant when a significant portion of max size. Logic ignores max size only for accounts that are privileged / system.

## Test scenarios

Checked values on accounts with NET or RAM that is:
a.) less than 1mb
b) less than 1gb but greater than 1mb
c.) greater than 1gb (eosio.evm)
d.) system account (eosio)

## Checklist:

-   [X] I have performed a self-review of my own code
-   [X] I have commented my code, particularly in hard-to-understand areas
-   [X] I have cleaned up the code in the areas my change touches
-   [X] My changes generate no new warnings
-   [X] I have checked my code and corrected any misspellings
-   [X] I have removed any unnecessary console messages
-   [X] I have tested for mobile functionality and responsiveness

telegram-cloud-photo-size-1-4994574957388344653-y

### CODE REVIEW

File: src/components/AccountCard.vue [hunk 1]
```diff
@@ -152,28 +156,56 @@ export default defineComponent({
             }
         };
 
-        const loadResources = () => {
-            ram_used.value = fixDec(
-                Number(accountData.value.ram_usage) / KILO_UNIT.value,
-            );
+        const determineRamUnit = (size: number) => {
```
Comment (donnyquixotic) [inline src/components/AccountCard.vue:159]:
```suggestion
        const determineUnit = (size: number) => {
```
method isn't specif to RAM (used for NET as well)

File: src/components/AccountCard.vue [hunk 2]
```diff
@@ -152,28 +156,56 @@ export default defineComponent({
...
+            return {
+                unit: 'kb',
+                denominator: KILO_UNIT.value,
+            };
+        };
 
+        const loadResources = () => {
+            let ramDenominator;
             if (props.account !== system_account.value) {
+                // in most cases the max value unit is more im portant than the
```
Comment (donnyquixotic) [inline src/components/AccountCard.vue:181]:
```suggestion
                // base displayed units on max value for readability 
```

File: src/components/AccountCard.vue [hunk 3]
```diff
@@ -152,28 +156,56 @@ export default defineComponent({
...
+                // used value unit. So we use the max value unit.
```
Comment (donnyquixotic) [inline src/components/AccountCard.vue:182]:
```suggestion
```

File: src/components/AccountCard.vue [hunk 4]
```diff
@@ -152,28 +156,56 @@ export default defineComponent({
...
+                // get max ram number
```
Comment (donnyquixotic) [inline src/components/AccountCard.vue:183]:
```suggestion
```

File: src/components/AccountCard.vue [hunk 5]
```diff
@@ -152,28 +156,56 @@ export default defineComponent({
...
+                const ramMaxNumber = Number(accountData.value.ram_quota);
+                // get denominator based on max number
```
Comment (donnyquixotic) [inline src/components/AccountCard.vue:185]:
```suggestion
```

File: src/components/AccountCard.vue [hunk 6]
```diff
@@ -152,28 +156,56 @@ export default defineComponent({
...
+                ramDenominator = determineRamUnit(ramMaxNumber).denominator;
```
Comment (donnyquixotic) [inline src/components/AccountCard.vue:186]:
```suggestion
                const ramUnitResult = determineUnit(ramMaxNumber);
                ramDenominator = ramUnitResult.denominator;
                ramUnit.value = ramUnitResult.unit;
```
You only need to call determineUnit once with the same input as the input value never changes.

File: src/components/AccountCard.vue [hunk 7]
```diff
@@ -152,28 +156,56 @@ export default defineComponent({
...
+                // get displayed number value
                 ram_max.value = fixDec(
-                    Number(accountData.value.ram_quota) / KILO_UNIT.value,
+                    ramMaxNumber / ramDenominator,
+                );
+                // get units
```
Comment (donnyquixotic) [inline src/components/AccountCard.vue:191]:
```suggestion
```

File: src/components/AccountCard.vue [hunk 8]
```diff
@@ -152,28 +156,56 @@ export default defineComponent({
...
+                ramUnit.value = determineRamUnit(ramMaxNumber).unit;
```
Comment (donnyquixotic) [inline src/components/AccountCard.vue:192]:
```suggestion
```
assigned on line 188 via suggesstion

File: src/components/AccountCard.vue [hunk 9]
```diff
@@ -152,28 +156,56 @@ export default defineComponent({
...
+                const netUsedNumber = Number(accountData.value.net_limit.used);
+                const netMaxNumber = Number(accountData.value.net_limit.max);
+                const netDenominator = determineRamUnit(netMaxNumber).denominator;
```
Comment (donnyquixotic) [inline src/components/AccountCard.vue:195]:
```suggestion
                const netUnitResult = determineUnit(netMaxNumber);
                const netDenominator = netUnitResult.denominator;
                netUnit.value = netUnitResult.unit;
```
You only need to call determineUnit once with the same input as the input value never changes.

File: src/components/AccountCard.vue [hunk 10]
```diff
@@ -152,28 +156,56 @@ export default defineComponent({
...
+                netUnit.value = determineRamUnit(netMaxNumber).unit;
```
Comment (donnyquixotic) [inline src/components/AccountCard.vue:196]:
```suggestion
```
refactored in previous suggestion

File: src/components/AccountCard.vue [hunk 11]
```diff
@@ -190,6 +222,15 @@ export default defineComponent({
                     stakedResources.value - stakedNET.value - stakedCPU.value,
                 );
             }
+            const ramUsedNumber = Number(accountData.value.ram_usage);
+            // only change denominator and units if wasn't already set
+            if (!ramDenominator) {
+                ramDenominator = determineRamUnit(ramUsedNumber).denominator;
```
Comment (donnyquixotic) [inline src/components/AccountCard.vue:228]:
```suggestion
                const ramUnitResult = determineUnit(ramUsedNumber);
                ramDenominator = ramUnitResult.denominator;
                ramUnit.value = ramUnitResult.unit;
```

File: src/components/AccountCard.vue [hunk 12]
```diff
@@ -190,6 +222,15 @@ export default defineComponent({
...
+                ramUnit.value = determineRamUnit(ramUsedNumber).unit;
```
Comment (donnyquixotic) [inline src/components/AccountCard.vue:229]:
```suggestion
```

File: src/components/AccountCard.vue [hunk 13]
```diff
@@ -375,6 +416,10 @@ export default defineComponent({
             accountPageSettings,
             MICRO_UNIT,
```
Comment (donnyquixotic) [inline src/components/AccountCard.vue:417]:
```suggestion
```
unnecessary to return, not referenced in template

File: src/components/AccountCard.vue [hunk 14]
```diff
@@ -375,6 +416,10 @@ export default defineComponent({
...
             KILO_UNIT,
```
Comment (donnyquixotic) [inline src/components/AccountCard.vue:418]:
```suggestion
```
unnecessary to return, not referenced in template

File: src/components/AccountCard.vue [hunk 15]
```diff
@@ -375,6 +416,10 @@ export default defineComponent({
...
+            MEGA_UNIT,
```
Comment (donnyquixotic) [inline src/components/AccountCard.vue:419]:
```suggestion
```
unnecessary to return, not referenced in template

File: src/components/AccountCard.vue [hunk 16]
```diff
@@ -375,6 +416,10 @@ export default defineComponent({
...
+            GIGA_UNIT,
```
Comment (donnyquixotic) [inline src/components/AccountCard.vue:420]:
```suggestion
```
unnecessary to return, not referenced in template
```

**Review by donnyquixotic — CHANGES_REQUESTED** [review]

**Comment by kylanhurt** [comment]
@donnyquixotic Comments handled

**Review by donnyquixotic — APPROVED** [review]

**Pull request merged**